src/helper/remote_config/protocol/tuf/parser.cpp

```diff
@@ -0,0 +1,392 @@
+
+    if (hashes_mapped.empty()) {
+        result.first =
+            remote_config_parser_result::hashes_path_targets_field_empty;
+        return result;
+    }
+
+    rapidjson::Value::ConstMemberIterator length_itr;
+    result.first = validate_field_is_present(target_itr, "length",
+        rapidjson::kNumberType, length_itr,
+        remote_config_parser_result::length_path_targets_field_missing,
+        remote_config_parser_result::length_path_targets_field_invalid);
+    if (result.first != remote_config_parser_result::success) {
+        return result;
+    }
+
+    std::string target_name(target_itr->name.GetString());
+    path path_object(v_itr->value.GetInt(), std::move(hashes_mapped),
+        length_itr->value.GetInt());
+    result.second.first = target_name;
+    result.second.second = path_object;
+    result.first = remote_config_parser_result::success;
+
+    return result;
+}
+
+std::pair<remote_config_parser_result, std::optional<targets>>
+parse_targets_signed(rapidjson::Value::ConstMemberIterator targets_signed_itr)
+{
+    std::pair<remote_config_parser_result, std::optional<targets>> result;
+    rapidjson::Value::ConstMemberIterator version_itr;
+    result.first = validate_field_is_present(targets_signed_itr, "version",
+        rapidjson::kNumberType, version_itr,
+        remote_config_parser_result::version_signed_targets_field_missing,
+        remote_config_parser_result::version_signed_targets_field_invalid);
+    if (result.first != remote_config_parser_result::success) {
+        return result;
+    }
+
+    rapidjson::Value::ConstMemberIterator targets_itr;
+    result.first = validate_field_is_present(targets_signed_itr, "targets",
+        rapidjson::kObjectType, targets_itr,
+        remote_config_parser_result::targets_signed_targets_field_missing,
+        remote_config_parser_result::targets_signed_targets_field_invalid);
+    if (result.first != remote_config_parser_result::success) {
+        return result;
+    }
+
+    std::vector<std::pair<std::string, std::optional<path>>> paths;
+    for (rapidjson::Value::ConstMemberIterator current_target =
+             targets_itr->value.MemberBegin();
+         current_target != targets_itr->value.MemberEnd(); ++current_target) {
+        auto [result_parse_target, p] = parse_target(current_target);
+        paths.push_back(p);
+        result.first = result_parse_target;
+        if (result.first != remote_config_parser_result::success) {
+            return result;
+        }
+    }
+
+    rapidjson::Value::ConstMemberIterator type_itr;
+    result.first = validate_field_is_present(targets_signed_itr, "_type",
+        rapidjson::kStringType, type_itr,
+        remote_config_parser_result::type_signed_targets_field_missing,
+        remote_config_parser_result::type_signed_targets_field_invalid);
+    if (result.first != remote_config_parser_result::success) {
+        return result;
+    }
+    if (strcmp(type_itr->value.GetString(), "targets") != 0) {
```

Comment:
```suggestion
    if (type_itr->value.GetString() == "targets"sv) {
```
You will also need to add `using namespace std::literals;` somewhere after your includes.

Reply:
Done